### PR TITLE
Ludwig CI for torch nightly

### PR DESCRIPTION
**Pull request by Infernaught**
Addresses the failing unit tests [here](https://github.com/ludwig-ai/ludwig/actions/runs/5522490134/jobs/10072155042?pr=3459). Does not address horovod issue.

### CODE REVIEW

**Comment by arnavgarg1** [comment]
Nice! do you also want to add Pytorch unpinning to this PR for our pytest suite?

File: ludwig/features/category_feature.py [hunk 1]
```diff
@@ -382,7 +382,7 @@ def output_shape(self) -> torch.Size:
         return torch.Size([1])
 
     def metric_kwargs(self):
-        return {"top_k": self.top_k, "num_classes": self.num_classes, "task": "multiclass"}
+        return {"top_k": self.top_k, "num_classes": self.num_classes}
```
Comment (justinxzhao) [inline ludwig/features/category_feature.py:385]: Why this change?

File: ludwig/utils/audio_utils.py [hunk 2]
```diff
@@ -64,8 +63,11 @@ def read_audio_from_path(path: str) -> Optional[TorchAudioTuple]:
 
     Useful for reading from a small number of paths. For more intensive reads, use backend.read_binary_files instead.
     """
-    bytes_obj = get_bytes_obj_from_path(path)
-    return read_audio_from_bytes_obj(bytes_obj)
+    try:
+        return torchaudio.backend.sox_io_backend.load(path)
+    except Exception as e:
+        logger.warning(e)
```
Comment (justinxzhao) [inline ludwig/utils/audio_utils.py:69]: I wonder if we should just raise an error.

@tgaddair @w4nderlust @connor-mccorm it feels like supporting invalid audio files just bites users later when they start investigating model quality. What do you think?

**Comment by justinxzhao** [comment]
> Nice! do you also want to add Pytorch unpinning to this PR for our pytest suite?

Agreed. @Infernaught could you apply https://github.com/ludwig-ai/ludwig/pull/3459 to your PR?
